frontend/src/pages/profile/components/achievement-list/achievement-list.tsx

```diff
@@ -0,0 +1,58 @@
+import { type FC } from 'react';
+
+import { BlockWithTooltip } from '~/libs/components/components.js';
+import { DataTooltipId } from '~/libs/enums/enums.js';
+import { getValidClassNames } from '~/libs/helpers/helpers.js';
+import {
+  Achievement,
+  AchievementTooltipContent,
+} from '~/pages/profile/components/components.js';
+import { type UserAchievement } from '~/pages/profile/libs/types/types.js';
+
+import styles from './styles.module.scss';
+
+type Properties = {
+  achievements: UserAchievement[];
+  shouldShowTooltip?: boolean;
+  className?: string;
+  classNameAchievement?: string;
+};
+
+const AchievementList: FC<Properties> = ({
+  achievements,
+  className,
+  classNameAchievement,
+  shouldShowTooltip = false,
+}) => {
+  return (
+    <ul className={getValidClassNames(className, styles.achievementList)}>
+      {achievements.map((achievement) => (
+        <li key={achievement.id}>
+          {shouldShowTooltip ? (
+            <BlockWithTooltip
+              tooltipContent={
+                <AchievementTooltipContent
+                  description={achievement.description}
+                  progress={achievement.progress}
+                />
+              }
+              tooltipId={DataTooltipId.ACHIEVEMENT_TOOLTIP}
+            >
+              <Achievement
+                achievement={achievement}
+                className={classNameAchievement}
+              />
+            </BlockWithTooltip>
+          ) : (
+            <Achievement
+              achievement={achievement}
+              className={classNameAchievement}
+            />
+          )}
```

Comment:
We can simplify this ternary by passing logic with tooltip blocking inside tooltip component
```suggestion
            <BlockWithTooltip
              tooltipContent={
                <AchievementTooltipContent
                  description={achievement.description}
                  progress={achievement.progress}
                />
              }
              tooltipId={DataTooltipId.ACHIEVEMENT_TOOLTIP}
              hasToShowTooltip={hasToShowTooltip}
            >
              <Achievement
                achievement={achievement}
                className={classNameAchievement}
              />
            </BlockWithTooltip>
```

Reply:
[872c299](https://github.com/BinaryStudioAcademy/bsa-2023-writorium/pull/156/commits/872c299d0e15e929c365b18e9248a493ea3455fc)